janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/JointIndexQuery.java

```diff
@@ -116,6 +116,7 @@ public JointIndexQuery updateLimit(int newLimit) {
             subqueries = new ArrayList<>(queries);
         }
         JointIndexQuery jointIndexQuery = new JointIndexQuery(subqueries);
+        jointIndexQuery.observeWith(this.profiler, false);
```

Comment:
When updating the limit and returning a new "cloned" query, we must also pass the profiler otherwise previous profiling results get lost.